lib/compile.js

```diff
@@ -87,7 +87,7 @@ function webpackCompile(config, logStats, ServerlessError) {
     _.forEach(stats, compileStats => {
       logStats(compileStats);
       if (compileStats.hasErrors()) {
-        throw new ServerlessError('Webpack compilation error, see stats above');
+        throw _.assign(new ServerlessError('Webpack compilation error, see stats above'), { stats: compileStats });
```

Comment:
This is not actually a change request, but rather grabbing the chance to discuss with people who cares about the project.

When I am experimenting the reverting of MultiCompile/ combined entries approach, I can't help but think of a complete rewrite of this project with TypeScript. The monkey patching approach brought down from Serverless V1 makes debugging and error tracing a bit tricky in many cases.

The effort is not small though, do you think it's worth doing?

Reply:
@vicary thanks for comment, but do you see this discussion specific to this contribution?

This PR is about grouping all compilation errors into one error, and not maintaining a situation where first one is reported, and others jump out aside (after Framework reports that command processing is finalized)

Reply:
I think it's more a global discussion not specific to your contribution

Reply:
Seeing that you're adding `stats` to the `ServerlessError` object, I just expanded the topic a bit.

But definitely not a blocker for this issue, just general discussion.